data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/S3ScanService.java

```diff
@@ -58,17 +67,30 @@ public S3ScanService(final S3SourceConfig s3SourceConfig,
         this.acknowledgementSetManager = acknowledgementSetManager;
         this.s3ObjectDeleteWorker = s3ObjectDeleteWorker;
         this.pluginMetrics = pluginMetrics;
+        this.executorService = Executors.newFixedThreadPool(s3SourceConfig.getNumWorkers(), BackgroundThreadFactory.defaultExecutorThreadFactory("s3-source-scan"));
     }
 
     public void start() {
-        scanObjectWorker = new ScanObjectWorker(s3ClientBuilderFactory.getS3Client(),
-                getScanOptions(),s3ObjectHandler,bucketOwnerProvider, sourceCoordinator, s3SourceConfig, acknowledgementSetManager, s3ObjectDeleteWorker, pluginMetrics);
-        scanObjectWorkerThread = new Thread(scanObjectWorker);
-        scanObjectWorkerThread.start();
+        int backOffMs = s3SourceConfig.getNumWorkers() > 1 ? FAST_BACKOFF_MILLIS : STANDARD_BACKOFF_MILLIS;
+        for (int i = 0; i < s3SourceConfig.getNumWorkers(); i++) {
+            ScanObjectWorker scanObjectWorker = new ScanObjectWorker(s3ClientBuilderFactory.getS3Client(),
+                    getScanOptions(),s3ObjectHandler,bucketOwnerProvider, sourceCoordinator, s3SourceConfig, acknowledgementSetManager, s3ObjectDeleteWorker, backOffMs, pluginMetrics);
+            executorService.submit(new Thread(scanObjectWorker));
+        }
     }
 
     public void stop() {
-        scanObjectWorker.stop();
+        executorService.shutdown();
```

Comment:
Let's make sure the scan workers are shutdown properly and that they give up their partition if they own one